dataland-frontend/src/components/resources/frameworkDataSearch/p2p/P2pPanel.vue

```diff
@@ -0,0 +1,332 @@
+            this.iteratesThroughSubcategories(
+              categoryObject,
+              categoryKey,
+              categoryResult,
+              dataIdOfP2pDataset,
+              listOfDataObjects
+            );
+
+            this.mapOfKpiKeysToDataObjectsArrays.set(categoryResult.label, listOfDataObjects);
+          }
+        });
+      }
+      this.listOfDataSetReportingPeriods = sortReportingPeriodsToDisplayAsColumns(
+        this.listOfDataSetReportingPeriods as ReportingPeriodOfDataSetWithId[]
+      );
+    },
+    /**
+     * Iterates through all subcategories of a category
+     * @param categoryObject the data object of the framework's category
+     * @param categoryKey the key of the corresponding framework's category
+     * @param categoryResult  the category object of the framework's category
+     * @param dataIdOfP2pDataset  the data ID of the P2P dataset
+     * @param listOfDataObjects a map containing the category and it's corresponding Kpis
+     */
+    iteratesThroughSubcategories(
+      categoryObject,
+      categoryKey,
+      categoryResult: Category,
+      dataIdOfP2pDataset: string,
+      listOfDataObjects: Array<KpiDataObject>
+    ) {
+      for (const [subCategoryKey, subCategoryObject] of Object.entries(categoryObject as object) as [
+        string,
+        object | null
+      ][]) {
+        if (subCategoryObject == null) continue;
+        this.assembleDataKpiObject(
+          subCategoryObject,
+          categoryKey,
+          subCategoryKey,
+          categoryResult,
+          dataIdOfP2pDataset,
+          listOfDataObjects
+        );
+      }
+    },
+    /**
+     * Builds the result Kpi Data Object and adds it to the result list
+     * @param subCategoryObject the data object of the framework's subcategory
+     * @param categoryKey the key of the corresponding framework's category
+     * @param subCategoryKey the key of the corresponding framework's subcategory
+     * @param categoryResult the category object of the framework's category
+     * @param dataIdOfP2pDataset the data ID of the P2P dataset
+     * @param listOfDataObjects a map containing the category and it's corresponding Kpis
+     */
+    assembleDataKpiObject(
+      subCategoryObject: object,
+      categoryKey,
+      subCategoryKey: string,
+      categoryResult: Category,
+      dataIdOfP2pDataset: string,
+      listOfDataObjects: Array<KpiDataObject>
+    ) {
+      for (const [kpiKey, kpiValue] of Object.entries(subCategoryObject) as [string, object] | null) {
+        if (kpiValue == null) continue;
+        const subcategory = assertDefined(
+          p2pDataModel
+            .find((category) => category.name === categoryKey)
+            ?.subcategories.find((subCategory) => subCategory.name === subCategoryKey)
+        );
+        const field = assertDefined(subcategory.fields.find((field) => field.name == kpiKey));
+        if (
+          this.p2pDataAndMetaInfo
```

Comment:
This very same thing is done for every dataset in the table. We could create a map as a computed property with a boolean as the showIf result with field.name as key

Reply:
Lets have a short call

Reply:
Paul and I did a small rework. Can you have a look at it if this fits?

Reply:
Können wir das nochmal besprechen?

Reply:
Klar, sobald dir das passt